quran/fields.py

```diff
@@ -0,0 +1,44 @@
+from django.core import exceptions
+from django.db import models
+
+
+class SegmentsField(models.TextField):
+    description = "A custom field to save list of timecodes in DB as a string"
+
+    def to_python(self, segments):
+        if isinstance(segments, list) or segments is None or segments == '':
+            return segments
+        try:
+            return [
+                tuple(int(timecode) for timecode in timecodes.split(':'))
+                for timecodes in segments.split(',')]
+        except (TypeError, ValueError):
+            raise exceptions.ValidationError(
+                message='Invalid string of segments',
+                code='invalid',
+            )
+
+    def from_db_value(self, value, expression, connection):
+        return self.to_python(value)
+
+    def get_prep_value(self, segments_list):
```

Comment:
```suggestion
    def get_prep_value(self, value):
```

Reply:
You should also call as a first line you parent's `get_prep_value` like this:
`super().get_prep_value()`

Reply:
There is some error If I add this line. Because `super.get_prep_value` runs `to_python`, that we don't need in our case. Because  `get_prep_value` is using to load data from python objects to DB string format, `to_python` from DB string to python objects. So I won't add it. Works fine without it.